### ISSUE TITLE
WebCrypto Support

### ISSUE THREAD

**Issue by ImSingee** [cloudflare/workers-rs#184]
I think it may be useful if support [Web Crypto](https://developers.cloudflare.com/workers/runtime-apis/web-crypto/) like [web_sys](https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.Crypto.html#)

**Comment by zebp** [comment]
I'm a little unsure if this will be worth it, I have two concerns.

1. We're trying to be as rust-y as possible, so calling into APIs that look exactly like the browser kind of breaks that illusion.
2. The bigger concern, is the overhead of calling into JS with all those copies vs just doing it in wasm worth it? Rust's crypto crates are usually pretty good but I'm not sure how fast they end up when compiled to wasm, I think we should do some benchmarks on this.

If we find that it probably isn't a good fit, users that do want to be able to use it should just be able to use web_sys.

**Comment by ImSingee** [comment]
In fact, I just can't find a way to use crypto method as easy as just using WebCrypto .

I found https://github.com/RustCrypto/block-ciphers , and because of the Warning section in their README, I'm afraid of using them.

I'm using [MagicCrypt](https://github.com/magiclen/rust-magiccrypt) now but it do too many things besides of just an encrypt/decrypt.

So I think, proving a way for using Web Crypto may be useful in some case.

**Comment by maxcountryman** [comment]
> In fact, I just can't find a way to use crypto method as easy as just using WebCrypto .

Have you given [ring](https://crates.io/crates/ring) a try?

I've found it to be excellent, performing well when compiled to WASM. It's my go-to for encryption in Rust along with argon2 (for e.g. password hashing).

**Comment by Perksey** [comment]
Ring is a great suggestion, but I still find myself needing WebCrypto to get access to a secure RNG.

**Comment by bugproof** [comment]
Any news?

https://developers.cloudflare.com/workers/runtime-apis/web-crypto/#_top

How do we use it?

**Comment by guybedford** [comment]
@bugproof yes, you can import `web_sys::Crypto` with the `Crypto` feature enabled for `web_sys` here. See https://wasm-bindgen.github.io/wasm-bindgen/api/web_sys/struct.Crypto.html.
